### ISSUE TITLE
Embed dissappears with Collpased topics course format

### ISSUE THREAD

**Issue by Jarvil** [gjb2048/moodle-format_topcoll#36]
Hello,

There is a problem with Collapsed topic and H5P Embed -feature (iframe).

Tested with Moodle 3.3, Mod_hvp 1.6 and format_topcoll 3.3.1.0 (2017042202) and Clean-theme

Steps to repeat;

Embed H5P activity to some section f.ex section 1
Click "close all" to fold all sections
Edit some random activity on the course (no need to edit it) and select "save and return to course"
Expand the section 1 and notice how the embedded H5P-activity isn't displayed, it only displays the bottom bar of the embedded H5P-activity
If you refresh the page with F5 then the embedded activity is displayed.
So the problem is that embedded H5P-activity isn't displayed if you fold all sections -> edit something -> return to the cousre frontpage until you manually refresh the page.

![h5p1](https://user-images.githubusercontent.com/15193418/34514836-f05e1522-f077-11e7-8acc-3b0c203a71e9.PNG)
![h5p2](https://user-images.githubusercontent.com/15193418/34514838-f0a1fada-f077-11e7-99d9-5d47f5d261a2.png)
![h5p3](https://user-images.githubusercontent.com/15193418/34514839-f0c18986-f077-11e7-8d3e-2686e5ab207f.png)
![h5p4](https://user-images.githubusercontent.com/15193418/34514841-f0e2b0b6-f077-11e7-841b-84c81b5762ab.png)
![h5p5](https://user-images.githubusercontent.com/15193418/34514844-f10006e8-f077-11e7-806b-494708ec6d61.png)
![h5p6](https://user-images.githubusercontent.com/15193418/34514846-f11d6418-f077-11e7-8b13-c8265cc48f33.PNG)
![h5p7](https://user-images.githubusercontent.com/15193418/34514847-f13986c0-f077-11e7-8cd0-101f4315de2e.PNG)
![h5p8](https://user-images.githubusercontent.com/15193418/34514848-f159133c-f077-11e7-8c3b-04849aa4dd6b.PNG)
![h5p9](https://user-images.githubusercontent.com/15193418/34514849-f1762328-f077-11e7-80ab-692ceed31ff6.PNG)
![h5p10](https://user-images.githubusercontent.com/15193418/34514850-f1a1af5c-f077-11e7-8744-29aceccc3bcd.PNG)
![h5p11](https://user-images.githubusercontent.com/15193418/34514851-f1c1adb6-f077-11e7-9acf-99988c7d38ce.PNG)

**Comment by gjb2048** [comment]
Noted, will look when I get time.

**Comment by Jarvil** [comment]
Great!

**Comment by gjb2048** [comment]
Yes I can replicate.  But there is a problem.  The H5P mod script is a dynamic one that orchestrates itself on page load to generate the markup required to the available space (there is a resize script to cope with window resize).  But if the script detects that the content is hidden (i.e. display: none) then it does not render it.  When the container 'display' is changed by CT then the script is unaware that it needs to recalculate things etc.

Looking at h5p-resizer.js with:

````
  // Let h5p iframes know we're ready!
  var iframes = document.getElementsByTagName('iframe');
  var ready = {
    context: 'h5p',
    action: 'ready'
  };
  for (var i = 0; i < iframes.length; i++) {
    if (iframes[i].src.indexOf('h5p') !== -1) {
      iframes[i].contentWindow.postMessage(ready, '*');
    }
  }
````

there might be a way of telling the H5P frames when the state changes and therefore to redraw.

**Comment by gjb2048** [comment]
Do you know?

To be continued....

**Comment by gjb2048** [comment]
Marked as an 'Enhancement' as not a fault with the code but integration with a 3rd party plugin.

**Comment by Jarvil** [comment]
Hello,

I'm sorry but I don't know H5P myself any better.

**Comment by Jarvil** [comment]
You should see;

https://github.com/h5p/h5p-moodle-plugin/issues/190

**Comment by gjb2048** [comment]
Is this still an issue?

**Comment by gjb2048** [comment]
@Jarvil Is this still an issue?

**Comment by gjb2048** [comment]
No more information forthcoming, closing.
